### PR TITLE
feat: support multiple time columns with time grain in Pivot Table v2

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

If the FF Generic Axis is enabled, the Pivot Table v2 will support multiple time columns with time grain. The `TIME GRAIN
` control also moves out from the time section.

There is a limitation in the current design that the Adhoc Column wasn't able to apply the Time Grain, I will open separated PR to support the dynamic detector for the column type.

#### After
The time grain is applied on `Order Date` and `Ship Date`
image

The time grain is applied on `Order Date` and `Ship Date` except `Category`

image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/plugins/plugin-chart-pivot-table/package.json [hunk 1]
```diff
@@ -25,7 +25,9 @@
   "publishConfig": {
     "access": "public"
   },
-  "dependencies": {},
+  "dependencies": {
+    "lodash": "^4.17.11"
```
Comment (kgabryje) [inline superset-frontend/plugins/plugin-chart-pivot-table/package.json:29]: can it be a peer dep?

File: superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts [hunk 2]
```diff
@@ -27,10 +33,29 @@ import { PivotTableQueryFormData } from '../types';
 export default function buildQuery(formData: PivotTableQueryFormData) {
   const { groupbyColumns = [], groupbyRows = [] } = formData;
   // TODO: add deduping of AdhocColumns
-  const groupbySet = new Set([
-    ...ensureIsArray<QueryFormColumn>(groupbyColumns),
-    ...ensureIsArray<QueryFormColumn>(groupbyRows),
-  ]);
+  const groupbySet = Array.from(
```
[the remark this answers is not in the document]
Comment (zhaoyongjie) [reply superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:36]: nice catch! will do

File: superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/controlPanel.tsx [hunk 3]
```diff
@@ -63,6 +69,57 @@ const config: ControlPanelConfig = {
...
           },
         ],
+        [
+          isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES)
+            ? {
+                name: 'time_grain_sqla',
+                config: {
+                  ...sharedControls.time_grain_sqla,
+                  visibility: ({ controls }) => {
+                    const DTTMLookup = Object.fromEntries(
```
Comment (kgabryje) [inline superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/controlPanel.tsx:79]:
```suggestion
                    const dttmLookup = Object.fromEntries(
```

File: superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts [hunk 4]
```diff
@@ -27,10 +33,29 @@ import { PivotTableQueryFormData } from '../types';
...
+    new Set([
+      ...ensureIsArray<QueryFormColumn>(groupbyColumns),
+      ...ensureIsArray<QueryFormColumn>(groupbyRows),
+    ]),
+  ).map(col => {
+    if (
+      isPhysicalColumn(col) &&
+      formData.time_grain_sqla &&
+      isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES) &&
+      formData?.datetimeColumnLookup[col]
```
Comment (kgabryje) [inline superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:46]: nit: do we need a null check here (we don't do null check in line 44). Maybe we should do `formData.datetimeColumnLookup?.[col]` instead?

Comment (zhaoyongjie) [reply superset-frontend/plugins/plugin-chart-pivot-table/src/plugin/buildQuery.ts:46]: done.

**Review by kgabryje — APPROVED** [review]
LGTM!

**Pull request merged**